File: Object-Oriented Programming/Python/Exercises/Ex5.py
print("Digite um numero inteiro: ")
num = input()

previous_prime = -1
is_prime = True

for i in range(int(num) - 1, 1, -1):
    for j in range(2, i):
        if(i % j == 0):
            is_prime = False
            break
        is_prime = True
    if(is_prime == True):
        previous_prime = i
        break
    
if(previous_prime != -1):
    print("O primeiro número primo menor que o número informado é: %d." % previous_prime)
else:
    print("Não existe número primo menor que o número informado.")

File: Object-Oriented Programming/Java/Exercises/List/src/Ex5.java
public class Ex5 {

	public static void main(String[] args) {
		System.out.printf("Digite um numero inteiro: ");
		int num;
		
		try {
			num = EntradaTeclado.leInt();
			} catch (Exception e) {
			 System.out.println("O número digitado não é valido.\n");
			 return;
			}

		int i, j;
		int previous_prime = -1;
		boolean is_prime = true;
		for (i = num - 1; i > 1; i--) {
			for (j = 2; j < i; j++) {
				if (i % j == 0) {
					is_prime = false;
					break;
				}
				is_prime = true;
			}
			if (is_prime == true) {
				previous_prime = i;
				break;
			}
		}
		
		if (previous_prime != -1) {
			System.out.printf("O primeiro número primo menor que o número informado é: %d.\n", previous_prime);
		} else {
			System.out.printf("Não existe número primo menor que o número informado.\n");
		}
	}
}
